### ISSUE TITLE
Add an e2e tests that detects error logs during etcd bootstrap

### ISSUE THREAD

**Issue by serathius** [etcd-io/etcd#17329]
### What would you like to be added?

Etcd should not write any error logs during normal operation. We could add a test that starts etcd cluster, does some simple operations and validates if there is any error level log written.

This could be further improved by enabling this check in all e2e tests when cluster/server is being closed. As some test intentionally inject errors, we could make it configurable option that is enabled by default and disabled in all tests we expect error to be written.

### Why is this needed?

Logs are used to provide useful debug information to users, however if mislabeled they could cause user to lose trust and ignore errors. For error logs to be useful they should only be emitted where there is something wrong.

We should prevent issues like https://github.com/etcd-io/etcd/issues/17245 where a change in code started generating a large number of error logs.

Proposed in https://github.com/etcd-io/etcd/pull/17249#issuecomment-1893399469 with agreement between maintainers.

**Comment by nitishfy** [comment]
Hi, I'd like to take this up!

**Comment by serathius** [comment]
Thanks, feel free to reach out to me on [slack](https://kubernetes.slack.com/archives/C3HD8ARJ5) ([get an invite](http://slack.kubernetes.io/)) if you need any help 
/assign @nitishfy

**Comment by serathius** [comment]
https://github.com/etcd-io/etcd/pull/17423 covered a single node cluster, we should also cover:
* 3 node cluster
* different TLS configurations.

**Comment by nitishfy** [comment]
> #17423 covered a single node cluster, we should also cover:
> 
> * 3 node cluster
> * different TLS configurations.

we can raise a new PR for it. WDYT?

**Comment by serathius** [comment]
SG, thanks

**Comment by serathius** [comment]
Also, when we are done, we should consider backporting the test.

**Comment by serathius** [comment]
Still remaining to be completed:
* Extend the [existing test](https://github.com/etcd-io/etcd/pull/17423) to also validate 3 node cluster. 
* Backport the test to release-3.5 and release-3.4 branches to prevent regressions there.

**Comment by ghouscht** [comment]
I think I can take this.

/assign

**Comment by ghouscht** [comment]
done 🙂 I think this can be closed now.

**Comment by serathius** [comment]
Awesome, thanks @ghouscht !

**Comment by ahrtr** [comment]
It looks like the test doesn't enable TLS by default. Should we also extend the test to support the case of enabling TLS?

**Comment by serathius** [comment]
Good point, I missed https://github.com/etcd-io/etcd/issues/17329#issuecomment-1943954845.

**Comment by ghouscht** [comment]
Oh I missed that as well, sorry! I'll have a look later on and will prepare another PR.

**Comment by ghouscht** [comment]
Something like this https://github.com/etcd-io/etcd/pull/18819? Or did you have something different in your mind?
